### PR TITLE
Feature/add file extension aliases 7365

### PR DESCRIPTION
**Pull request by Aditya-PS-05**
## Summary
This pull request adds support for additional file extension aliases in the SourceDistExtension and ExtensionError enums. The newly supported file extensions include .tbz, .tgz, .txz, .tar.lz, .tar.lzma. These changes align the extensions supported by the SourceDistExtension with those used in Python packaging tools, enhancing compatibility with a broader range of source distribution formats.

closes #7201 

## Test Plan
should be added or updated to verify that the new extensions are correctly recognized as valid source distributions and that errors are correctly raised when unsupported extensions are provided.

### CODE REVIEW

File: crates/distribution-filename/src/extension.rs [hunk 1]
```diff
@@ -68,17 +68,18 @@ impl SourceDistExtension {
         };
 
         match extension {
-            "zip" => Ok(Self::Zip),
+            "zip"=> Ok(Self::Zip),
             "gz" if is_tar(path.as_ref()) => Ok(Self::TarGz),
-            "tgz" => Ok(Self::TarGz),
-            "bz2" if is_tar(path.as_ref()) => Ok(Self::TarBz2),
-            "xz" if is_tar(path.as_ref()) => Ok(Self::TarXz),
+            "tgz" | "tar" => Ok(Self::TarGz), // `.tar` included here
```
Comment (charliermarsh) [inline crates/distribution-filename/src/extension.rs:73]: It might be that `.tar` is just uncompressed tar without gzip?

Comment (Aditya-PS-05) [reply crates/distribution-filename/src/extension.rs:73]: Thanks for suggestion.

Comment (Aditya-PS-05) [reply crates/distribution-filename/src/extension.rs:73]: Should I need to update the SourceDistExtension enum?

Comment (charliermarsh) [reply crates/distribution-filename/src/extension.rs]: I believe so, yeah.

Comment (charliermarsh) [reply crates/distribution-filename/src/extension.rs]: Do you want some help finishing this off?

Comment (Aditya-PS-05) [reply crates/distribution-filename/src/extension.rs]: Yeah I need some help, 

in extension.rs file: 
```
pub enum SourceDistExtension {
    Zip,
    TarGz,
    TarBz2,
    TarXz,
    Tar,
    TarZst,
    TarLz,
    TarLzma,
}
```

```
match extension {
            "zip" => Ok(Self::Zip),
            "gz" if is_tar(path.as_ref()) => Ok(Self::TarGz),
            "tgz" => Ok(Self::TarGz),
            "tar" => Ok(Self::Tar),
            "bz2" | "tbz" if is_tar(path.as_ref()) => Ok(Self::TarBz2),
            "xz" | "txz" | "tlz" | "tar.lz" | "tar.lzma" if is_tar(path.as_ref()) => {
                Ok(Self::TarXz)
            }
            "zst" if is_tar(path.as_ref()) => Ok(Self::TarZst),
            _ => Err(ExtensionError::SourceDist),
        }
```
```
impl Display for SourceDistExtension {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        match self {
            Self::Zip => f.write_str("zip"),
            Self::TarGz => f.write_str("tar.gz"),
            Self::TarBz2 => f.write_str("tar.bz2"),
            Self::TarXz => f.write_str("tar.xz"),
            Self::Tar => f.write_str("tar"),
            Self::TarZst => f.write_str("tar.zst"),
            Self::TarLz => f.write_str("tar.lz"),
            Self::TarLzma => f.write_str("tar.lzma"),
        }
    }
}

// #[derive(Error, Debug)]
pub enum ExtensionError {
    #[error("`.whl`, `.zip`, `.tar.gz`, `.tgz`, `.tar.bz2`, `.tbz`, `.tar.xz`, `.txz`, `.tar.lz`, `.tar.lzma`, `.tar.zst`")]
    Dist,
    #[error("`.zip`, `.tar.gz`, `.tgz`, `.tar.bz2`, `.tbz`, `.tar.xz`, `.txz`, `.tar.lz`, `.tar.lzma`, `.tar.zst`")]
    SourceDist,
}

```

Where am I going wrong, as some tests are being failed. I tried to took inspiration from the issue #7201 as he done the same thing.

**Pull request closed without merging**